cluster-autoscaler/core/scale_up.go

```diff
@@ -190,6 +233,37 @@ func ScaleUp(context *AutoscalingContext, unschedulablePods []*apiv1.Pod, nodes
 			}
 		}
 
+		nodeInfo, found := nodeInfos[bestOption.NodeGroup.Id()]
```

Comment:
separate function

Reply:
Done

Reply:
Moved most of the logic to a separate function

Reply:
Done.